### PR TITLE
Draft/commands component

### PR DESCRIPTION
**Pull request by matipojo**

### CODE REVIEW

File: assets/dev/js/modules/imports/component.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+		return this.defaultRoute;
+	}
+
+	removeTab( tab ) {
+		delete this.tabs[ tab ];
+	}
+
+	addTab( tab, title, position ) {
+		this.tabs[ tab ] = title;
+		if ( undefined !== typeof position ) {
```
Comment (kobizz) [inline assets/dev/js/modules/imports/component.js:125]: You can just write `if ( position )`

File: assets/dev/js/editor/components/hotkeys/component.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+export default class extends elementorModules.Component {
+	constructor( ...args ) {
+		super( ...args );
+
+		this.title = 'Shortcuts';
```
Comment (kobizz) [inline assets/dev/js/editor/components/hotkeys/component.js]: Is it not supposed to be translated?

File: assets/dev/js/editor/components/hotkeys/component.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+	open() {
+		this.parent.layout.showModal();
+		return true;
+	}
+
+	getRoutes() {
+		return {
+			'': () => {},
+		};
```
Comment (kobizz) [inline assets/dev/js/editor/components/hotkeys/component.js:17]: What is?

File: assets/dev/js/editor/components/hotkeys/component.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+	init( args ) {
+		super.init( args );
+
+		this.parent.layout.getModal().on( 'hide', () => elementorCommon.route.close( 'shortcuts' ) );
+	}
+
+	open() {
+		this.parent.layout.showModal();
+		return true;
```
Comment (kobizz) [inline assets/dev/js/editor/components/hotkeys/component.js]: Why is the returning value required?

File: assets/dev/js/editor/components/hotkeys/hotkeys.js [hunk 5]
```diff
@@ -1,11 +1,10 @@
...
+import Component from './component';
 
 export default class extends elementorModules.Module {
 	onInit() {
 		this.layout = new ModalLayout();
 
-		elementorCommon.route.register( 'shortcuts', () => this.layout.showModal(), { keys: 'ctrl+?' } );
-
-		this.layout.getModal().on( 'hide', () => elementorCommon.route.close( 'shortcuts' ) );
+		elementorCommon.components.register( new Component(), { parent: this } );
```
Comment (kobizz) [inline assets/dev/js/editor/components/hotkeys/hotkeys.js]: I'm not sure if `parent` is the right term

File: assets/dev/js/editor/components/settings/general/component.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+export default class extends elementorModules.Component {
+	constructor( ...args ) {
+		super( ...args );
+
+		this.title = 'Global Settings';
```
Comment (kobizz) [inline assets/dev/js/editor/components/settings/general/component.js]: Same as above

File: assets/dev/js/editor/editor.js [hunk 7]
```diff
@@ -144,7 +145,7 @@ const App = Marionette.Application.extend( {
 			ignore: '.elementor-control-dynamic-switcher',
 		},
 		panelFooterSubMenus: {
-			element: '.elementor-panel-footer-tool',
+			element: '.elementor-panel-footer-tool.elementor-toggle-state',
```
Comment (kobizz) [inline assets/dev/js/editor/editor.js:152]: What is?

File: assets/dev/js/editor/elements/component.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
+// TODO: Maybe it should merge with panel/elements?
```
Comment (kobizz) [inline assets/dev/js/editor/elements/component.js:1]: What is this component responsible for?

File: assets/dev/js/editor/regions/navigator/component.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+		this.parent.open();
+		return true;
+	}
+
+	close() {
+		this.parent.close();
+	}
+
+	getRoutes() {
+		return {
```
Comment (kobizz) [inline assets/dev/js/editor/regions/navigator/component.js:7]: ?

File: assets/dev/js/editor/regions/panel/pages/elements/component.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
+export default class extends elementorModules.Component {
```
Comment (kobizz) [inline assets/dev/js/editor/regions/panel/pages/elements/component.js:1]: I'm not sure all of these components are really necessary

File: assets/dev/js/modules/imports/component.js [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+	getTabsWrapperSelector() {
+		return '';
+	}
+
+	getTabRoute( tab ) {
+		return this.namespace + '/' + tab;
+	}
+
+	activateTab( tab, routeArgs ) {
+		if ( this.tabRenderer ) {
```
Comment (kobizz) [inline assets/dev/js/modules/imports/component.js]: Needs more clarification

File: core/common/assets/js/utils/shortcuts.js [hunk 12]
```diff
@@ -59,8 +59,16 @@ export default class Shortcuts {
 		}
 
 		jQuery.each( handlers, ( key, handler ) => {
-			if ( handler.component && handler.component !== this.component ) {
-				return;
+			if ( handler.scope ) {
+				const InScope = handler.scope.some( ( scope ) => {
```
Comment (kobizz) [inline core/common/assets/js/utils/shortcuts.js]: `inScope`

File: core/common/modules/finder/assets/js/component.js [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+		this.parent.layout.getModal().on( 'hide', () => elementorCommon.route.close( this.namespace ) );
+	}
+
+	open() {
+		this.parent.layout.showModal();
+		return true;
+	}
+
+	getRoutes() {
+		return {
```
Comment (kobizz) [inline core/common/modules/finder/assets/js/component.js:15]: ?

File: includes/editor-templates/panel-elements.php [hunk 14]
```diff
@@ -10,8 +10,8 @@
 		<i class="eicon-loading eicon-animation-spin"></i>
 	</div>
 	<div id="elementor-panel-elements-navigation" class="elementor-panel-navigation">
```
Comment (kobizz) [inline includes/editor-templates/panel-elements.php:12]: Why?

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/modules/imports/component.js (hunk 1 above)
Comment (matipojo) [reply assets/dev/js/modules/imports/component.js:125]: It can be 0.

File: core/common/assets/js/utils/components.js [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+	}
+
+	printAll() {
+		console.log( Object.keys( this.components ).sort() ); // eslint-disable-line no-console
+	}
+
+	register( component, args ) {
+		this.components[ component.namespace ] = component;
+
+		component.init( args );
```
Comment (kobizz) [inline core/common/assets/js/utils/components.js]: Use `__construct`

File: assets/dev/js/editor/components/hotkeys/hotkeys.js (hunk 5 above)
Comment (kobizz) [reply assets/dev/js/editor/components/hotkeys/hotkeys.js]: context

File: assets/dev/js/editor/editor.js (hunk 7 above)
Comment (kobizz) [reply assets/dev/js/editor/editor.js:152]: Remove CSS

File: modules/history/assets/js/module.js [hunk 16]
```diff
@@ -14,6 +15,11 @@ Manager = function() {
 	var init = function() {
 		elementor.on( 'preview:loaded', addPanelPage );
 
+		elementorCommon.components.register( new Component(), {
+			parent: self,
+			tabRenderer: ( tab ) => elementor.getPanelView().setPage( 'historyPage' ).showView( tab ),
```
Comment (kobizz) [inline modules/history/assets/js/module.js]: `renderTab`

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/editor.js (hunk 7 above)
Comment (matipojo) [reply assets/dev/js/editor/editor.js:152]: Nothing to remove

**Pull request closed without merging**